### PR TITLE
UnusedPublicClassMethodRule: fix possible crash

### LINKED ISSUE #89: crash
running unused-public on 

https://github.com/rectorphp/rector-src/blob/7b7b4fb61bc2a7088823e7cfabf8a29b4fec516c/rules-tests/CodingStyle/Rector/FuncCall/CallUserFuncToMethodCallRector/Fixture/skip_string_variable.php.inc

leads to a crash

```
Internal error: Internal error: Internal error. in file                                                                                                                        
     /Users/staabm/workspace/rector-src/rules-tests/CodingStyle/Rector/FuncCall/CallUserFuncToMethodCallRector/Fixture/skip_string_variable.php.inc                                 
                                                                                                                                                                                    
     Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                                                      
     #0 /Users/staabm/workspace/rector-src/vendor/tomasvotruba/unused-public/src/Collectors/Callable_/CallUserFuncCollector.php(88):                                                
     PHPStan\Type\Constant\ConstantArrayTypeAndMethod->getType()                                                                                                                    
     #1 phar:///Users/staabm/workspace/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(128):                                                           
     TomasVotruba\UnusedPublic\Collectors\Callable_\CallUserFuncCollector->processNode(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                
     #2 phar:///Users/staabm/workspace/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassStatementsGatherer.php(108):                                                    
     PHPStan\Analyser\FileAnalyser->PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                                        
     #3 phar:///Users/staabm/workspace/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(539):                                                      
     PHPStan\Node\ClassStatementsGatherer->__invoke(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                                                   
     #4 phar:///Users/staabm/workspace/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(2631):                                                     
     PHPStan\Analyser\NodeScopeResolver::PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope))                                   
     #5 phar:///Users/staabm/workspace/rector-src/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(1570):                                                     
     PHPStan\Analyser\NodeScopeResolver->callNodeCallbackWithExpression(Object(Closure), Object(PhpParser\Node\Expr\FuncCall), Object(PHPStan\Analyser\MutatingScope),              
     Object(PHPStan\Analyser\ExpressionContext))       
```

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/TomasVotruba/unused-public/issues/89

### CODE REVIEW

File: src/Collectors/Callable_/CallUserFuncCollector.php [hunk 1]
```diff
@@ -82,4 +76,22 @@ public function processNode(Node $node, Scope $scope): ?array
 
         return $classMethodReferences;
     }
+
+    private function shouldSkipNode(FuncCall $node): bool {
```
Comment (staabm) [inline src/Collectors/Callable_/CallUserFuncCollector.php]: refactor to reduce complexity

File: src/Collectors/Callable_/CallUserFuncCollector.php [hunk 2]
```diff
@@ -73,6 +63,10 @@ public function processNode(Node $node, Scope $scope): ?array
 
         $classMethodReferences = [];
         foreach ($typeAndMethodNames as $typeAndMethodName) {
+            if ($typeAndMethodName->isUnknown()) {
```
Comment (staabm) [inline src/Collectors/Callable_/CallUserFuncCollector.php:71]: the actual fix

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Comment by staabm** [comment]
could you please tag a new release?

**Pull request merged**